File: Scripts/evaluations/rq3_userstudy/user2/TestCases/2/PackageDelivery/SendParcel_Parcel_sizeScreen.js
import React from 'react';

import {

  View,

  Text,

  StyleSheet,

  SafeAreaView,

  ScrollView,

  TouchableOpacity,

  Image,

} from 'react-native';

import { useNavigation } from '@react-navigation/native';



const SendParcel_Parcel_sizeScreen = () => {

  const navigation = useNavigation();



  const navigateToMyParcels = () => {

    navigation.navigate('Home_Top_drawer_close');

  };



  const navigateToParcelCenter = () => {

    navigation.navigate('Parcel_center');

  };



  const ParcelSizeOption = ({ size, maxDimensions, description, icon }) => (

    <TouchableOpacity style={styles.parcelSizeOption}>

      <Image source={require('../assets/snack-icon.png')} style={styles.parcelIcon} />

      <View style={styles.parcelSizeInfo}>

        <Text style={styles.parcelSizeTitle}>{size}</Text>

        <Text style={styles.parcelSizeMaxDimensions}>{maxDimensions}</Text>

        <Text style={styles.parcelSizeDescription}>{description}</Text>

      </View>

    </TouchableOpacity>

  );



  return (

    <SafeAreaView style={styles.container}>

      <ScrollView style={styles.scrollView}>

        <Text style={styles.title}>Send parcel</Text>

        <Text style={styles.subtitle}>Parcel size</Text>



        <View style={styles.parcelSizesContainer}>

          <ParcelSizeOption

            size="Small"

            maxDimensions="Max. 25 kg, 8 x 38 x 64 cm"

            description="Fits in an envelope"

          />

          <ParcelSizeOption

            size="Medium"

            maxDimensions="Max. 25 kg, 19 x 38 x 64 cm"

            description="Fits in a shoe box"

          />

          <ParcelSizeOption

            size="Large"

            maxDimensions="Max. 25 kg, 41 x 38 x 64 cm"

            description="Fits in a cardboard box"

          />

          <ParcelSizeOption

            size="Custom"

            maxDimensions="Max: 30kg or 300cm"

            description="Fits on a skid"

          />

        </View>

      </ScrollView>



      <View style={styles.footer}>

        <TouchableOpacity style={styles.footerButton} onPress={navigateToMyParcels}>

          <Image source={require('../assets/snack-icon.png')} style={styles.footerIcon} />

          <Text style={styles.footerButtonText}>My parcels</Text>

        </TouchableOpacity>

        <TouchableOpacity style={styles.footerButton}>

          <Image source={require('../assets/snack-icon.png')} style={styles.footerIcon} />

          <Text style={styles.footerButtonTextActive}>Send parcel</Text>

        </TouchableOpacity>

        <TouchableOpacity style={styles.footerButton} onPress={navigateToParcelCenter}>

          <Image source={require('../assets/snack-icon.png')} style={styles.footerIcon} />

          <Text style={styles.footerButtonText}>Parcel center</Text>

        </TouchableOpacity>

      </View>

    </SafeAreaView>

  );

};



const styles = StyleSheet.create({

  container: {

    flex: 1,

    backgroundColor: '#FFFFFF',

  },

  scrollView: {

    flex: 1,

    padding: 20,

  },

  title: {

    fontSize: 32,

    fontWeight: 'bold',

    marginBottom: 10,

  },

  subtitle: {

    fontSize: 24,

    fontWeight: 'bold',

    marginBottom: 20,

  },

  parcelSizesContainer: {

    backgroundColor: '#F5F5F5',

    borderRadius: 10,

    padding: 15,

  },

  parcelSizeOption: {

    flexDirection: 'row',

    alignItems: 'center',

    marginBottom: 20,

  },

  parcelIcon: {

    width: 60,

    height: 60,

    marginRight: 15,

  },

  parcelSizeInfo: {

    flex: 1,

  },

  parcelSizeTitle: {

    fontSize: 18,

    fontWeight: 'bold',

    marginBottom: 5,

  },

  parcelSizeMaxDimensions: {

    fontSize: 14,

    marginBottom: 5,

  },

  parcelSizeDescription: {

    fontSize: 14,

    color: '#888888',

  },

  footer: {

    flexDirection: 'row',

    justifyContent: 'space-around',

    borderTopWidth: 1,

    borderTopColor: '#E0E0E0',

    paddingVertical: 10,

  },

  footerButton: {

    alignItems: 'center',

  },

  footerIcon: {

    width: 24,

    height: 24,

    marginBottom: 5,

  },

  footerButtonText: {

    fontSize: 12,

    color: '#888888',

  },

  footerButtonTextActive: {

    fontSize: 12,

    color: '#000000',

    fontWeight: 'bold',

  },

});



export default SendParcel_Parcel_sizeScreen;

File: Scripts/results/generated_code_results/Flutter/bak_code/Utility&PracticalTools/PackageDelivery/SendParcel_Parcel_sizeScreen.dart
import 'package:flutter/material.dart';
class SendParcel_Parcel_sizeScreen extends StatefulWidget {
@override
_SendParcel_Parcel_sizeScreenState createState() => _SendParcel_Parcel_sizeScreenState();
}
class _SendParcel_Parcel_sizeScreenState extends State<SendParcel_Parcel_sizeScreen> {
String selectedSize = '';
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
title: Text('Send parcel'),
backgroundColor: Colors.white,
foregroundColor: Colors.black,
),
body: SingleChildScrollView(
child: Padding(
padding: const EdgeInsets.all(16.0),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text(
'Parcel size',
style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
),
SizedBox(height: 16),
_buildParcelSizeCard(
'Small',
'Max. 25 kg, 8 x 38 x 64 cm',
'Fits in an envelope',
'assets/images/template.png',
),
_buildParcelSizeCard(
'Medium',
'Max. 25 kg, 19 x 38 x 64 cm',
'Fits in a shoe box',
'assets/images/template.png',
),
_buildParcelSizeCard(
'Large',
'Max. 25 kg, 41 x 38 x 64 cm',
'Fits in a cardboard box',
'assets/images/template.png',
),
_buildParcelSizeCard(
'Custom',
'Max: 30kg or 300cm',
'Fits on a skid',
'assets/images/template.png',
),
],
),
),
),
bottomNavigationBar: BottomNavigationBar(
currentIndex: 1,
items: [
BottomNavigationBarItem(
icon: Icon(Icons.mail_outline),
label: 'My parcels',
),
BottomNavigationBarItem(
icon: Icon(Icons.send),
label: 'Send parcel',
),
BottomNavigationBarItem(
icon: Icon(Icons.location_on_outlined),
label: 'Parcel center',
),
],
),
floatingActionButton: FloatingActionButton.extended(
onPressed: () {
if (selectedSize.isNotEmpty) {
// TODO: Navigate to the next screen or process the parcel
ScaffoldMessenger.of(context).showSnackBar(
SnackBar(content: Text('Parcel size selected: $selectedSize')),
);
} else {
ScaffoldMessenger.of(context).showSnackBar(
SnackBar(content: Text('Please select a parcel size')),
);
}
},
label: Text('Next'),
icon: Icon(Icons.arrow_forward),
backgroundColor: Colors.black,
),
);
}
Widget _buildParcelSizeCard(String title, String dimensions, String description, String imagePath) {
return GestureDetector(
onTap: () {
setState(() {
selectedSize = title;
});
},
child: Card(
margin: EdgeInsets.only(bottom: 16),
shape: RoundedRectangleBorder(
borderRadius: BorderRadius.circular(8),
side: BorderSide(
color: selectedSize == title ? Colors.black : Colors.grey.shade300,
width: 2,
),
),
child: Padding(
padding: const EdgeInsets.all(16.0),
child: Row(
children: [
Image.asset(imagePath, width: 60, height: 60),
SizedBox(width: 16),
Expanded(
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text(title, style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),
Text(dimensions),
Text(description, style: TextStyle(color: Colors.grey)),
],
),
),
if (selectedSize == title)
Icon(Icons.check_circle, color: Colors.black),
],
),
),
),
);
}
}